compiler/optimizer/Inliner.cpp

```diff
@@ -2617,7 +2617,8 @@ TR_TransformInlinedFunction::transform()
    // If the first block has exception predecessors or multiply predecessors then we can't merge
    // the first block with the caller's block
    //
-   if ((firstBlock->getPredecessors().size() > 1) ||
+   if (comp()->isJProfilingCompilation() ||
+       (firstBlock->getPredecessors().size() > 1) ||
```

Comment:
This code controls whether we will merge the method entry block of an inlined method with the caller's block. When running profiling, this causes confusion when profiling owing to the fact we cannot see the entry block of the inlined method. This change forces us to split the block which fixes this confusion quite neatly.